### PR TITLE
vm: add initial interpreter implementation

### PR DESCRIPTION
**Pull request by lheeger-jump**
This adds the initial actual interpreter for the SBPF VM.

### CODE REVIEW

File: src/vm/fd_vm_interp_dispatch_tab.c [hunk 1]
```diff
@@ -0,0 +1,406 @@
+/* This is the BPF interpreter dispatch table.
+ * Each instruction is implemented in this table.
+ *
+ * See `fd_sbpf_interp.c` for usage.
```
Comment (lheeger-jump) [inline src/vm/fd_vm_interp_dispatch_tab.c:4]:
```suggestion
 * See `fd_vm_interp.c` for usage.
```

File: src/vm/fd_vm_interp_locs.c [hunk 2]
```diff
@@ -0,0 +1,228 @@
+/* This is the sBPF interpreter jump location table. It is an array where each index is an
+ * opcode that can be jumped to be executed.
+ *  
+ * See usage in `fd_sbpf_interp.c`
```
Comment (lheeger-jump) [inline src/vm/fd_vm_interp_locs.c:4]:
```suggestion
 * See usage in `fd_vm_interp.c`
```

[the remark this answers is not in the document]
Comment (lheeger-jump) [reply src/vm/fd_vm_interp_locs.c:4]: Done

File: config/with-brutality.mk [hunk 3]
```diff
@@ -2,7 +2,6 @@ CPPFLAGS+=            \
   -Werror             \
   -Wall               \
   -Wextra             \
-  -Wpedantic          \
   -Wstrict-aliasing=2 \
```
[the remark this answers is not in the document]
Comment (lheeger-jump) [reply config/with-brutality.mk:5]: What is the recommended way to do that?

[the remark this answers is not in the document]
Comment (jumpsiegel) [reply config/with-brutality.mk:5]: #pragma GCC diagnostic push          
#pragma GCC diagnostic ignored "-Wpedantic"    
 ... code that was tripping the warning
#pragma GCC diagnostic pop

**Pull request merged**